jax/_src/numpy/lax_numpy.py

```diff
@@ -2271,8 +2271,16 @@ def identity(n: DimSize, dtype: Optional[DTypeLike] = None) -> Array:
 def arange(start: DimSize, stop: Optional[DimSize] = None,
            step: Optional[DimSize] = None, dtype: Optional[DTypeLike] = None) -> Array:
   dtypes.check_user_dtype_supported(dtype, "arange")
-  require = partial(core.concrete_or_error, None)
-  msg = "It arose in jax.numpy.arange argument `{}`.".format
+  start_name = "stop" if stop is None and step is None else "start"
+  if not jax.config.jax_dynamic_shapes:
+    util.check_arraylike("arange", start)
+    start = core.concrete_or_error(None, start, f"It arose in the jnp.arange argument '{start_name}'")
```

Comment:
This is still performing string interpolation.

Reply:
Sure, but it's pretty safe, no? Would you prefer I put the if statement here and only call `core.concrete_or_error` with string literals?

Reply:
OK, I've removed string interpolation from this error message.